packages/material-ui-types/index.d.ts

```diff
@@ -38,9 +38,31 @@ export type PropInjector<InjectedProps, AdditionalProps = {}> = <
  */
 export type Omit<T, K extends keyof any> = T extends any ? Pick<T, Exclude<keyof T, K>> : never;
 
+/**
+ * Generate a set of string literal types with the given default record `T` and
+ * override record `U`.
+ *
+ * If the property value was `true`, the property key will be added to the
+ * string union.
+ *
+ * @internal
+ */
+export type OverridableStringUnion<T, U = {}> = GenerateStringUnion<Overwrite<T, U>>;
+
 /**
  * Like `T & U`, but using the value types from `U` where their properties overlap.
  *
  * @internal
  */
 export type Overwrite<T, U> = Omit<T, keyof U> & U;
+
+type GenerateStringUnion<T> = Extract<
+  {
+    [Key in keyof T]: T[Key] extends never
+      ? never
+      : Exclude<T[Key], true> extends never
```

Comment:
Do we need this condition or can we just assume that if `T[Key] === never` => omit `Key`. I guess you want to support `{ xs: false }` but I want to be cautious with conditional types regarding perf.

Reply:
@eps1lon This condition type ensures only property keys with the value of `true` will be added to the union type, other values (like `never` or `true | false`) will act as `false`.

If my method affects performance, I'm okay to use only `never` to remove default values.

Reply:
I'd say we flip this around and include if it `true extends T[Key]` and omit anything else. That way we keep the current behavior but have one less conditional.

Reply:
> I'd say we flip this around and include if it `true extends T[Key]` and omit anything else. That way we keep the current behavior but have one less conditional.

@eps1lon Oh, I never thought about it approach...

Reply:
Oops, there is a problem with that approach:

```typescript
// true extends T[Key]

type T = true extends true | false ? true : false;
// > true

type T = true extends true & false ? true : false;
// > false

// T[Key] extends true

type T = true | false extends true ? true : false;
// > false

type T = true & false extends true ? true : false;
// > true
```

Reply:
I wouldn't worry about it. If you write `interface OverrideBreakpoints { xs: true | false }` then this is already confusing.